### ISSUE TITLE
Feature Request: Log Lists

### ISSUE THREAD

**Issue by cemde** [neptune-ai/neptune-client#1173]
## Motivation

In my hyperparameters, I have a list of values I'd like to log as such. E.g.

```yaml
lr_scheduler:
    gamma: 0.1
    at: [10, 20, 30]
```

i.e. a LR scheduler that reduces the LR at steps 10, 20, and 30. When I try to log a list such as `at`, I get an error message:
```python
NeptuneDeprecationWarning: The object you're logging will be implicitly cast to a string. We'll end support of this behavior in `neptune-client==1.0.0`. To log the object as a string, use `str(object)` instead.
```

## Issue
Objects of type `list` is casted to `str` and then logged as `"[...,...,...]"`

## Proposed Solution
Allow logging of `list` and `tuple` as such.

**Comment by Blaizzy** [comment]
Hi @cemde 

Prince here,

Thanks for reaching out!

Could you share a minimal reproducible code example?

**Comment by cemde** [comment]
## Code
```python
import neptune.new as neptune

run = neptune.init_run(project="User/Project")

hparams = {
    "optim": {
        "scheduler": {
            "gamma": 0.1,
            "at": [10,20,30]
        }
    }
}

print(hparams)
run["train/parameters"] = hparams

run.stop()
```

## stdout
```
https://app.neptune.ai/User/Project/e/P-1
{'optim': {'scheduler': {'gamma': 0.1, 'at': [10, 20, 30]}}}
/.../.conda/envs/jaxnew/lib/python3.10/site-packages/neptune/new/attributes/attribute.py:64: NeptuneDeprecationWarning: The object you're logging will be implicitly cast to a string. We'll end support of this behavior in `neptune-client==1.0.0`. To log the object as a string, use `str(object)` instead.
  return self.assign(value, wait)
Shutting down background jobs, please wait a moment...
Done!
Waiting for the remaining 44 operations to synchronize with Neptune. Do not kill this process.
All 44 operations synced, thanks for waiting!
Explore the metadata in the Neptune app:
https://app.neptune.ai/User/Project/e/P-1
```

**Comment by Blaizzy** [comment]
In your case, if you pass the entire dictionary, we currently will cast every object to a string but starting neptune-client v1.0.0 you will get an error message.

On the other hand, if you want to keep the datatype(i.e. list) you can log it as a FloatSeries into the same namespace.

```python
import neptune.new as neptune
from neptune.new.types import FloatSeries

run = neptune.init_run(project="User/Project")

hparams = {
    "optim": {
        "scheduler": {
            "gamma": 0.1,
            "at": FloatSeries([10,20,30])
        }
    }
}

print(hparams)
run["train/parameters"] = hparams

run.stop()

```

Docs: https://docs.neptune.ai/api/field_types/#floatseries

**Comment by cemde** [comment]
thanks. that will probably work. It would be nice if the conversion to float / int / str series would be the default behaviour (if there isnt a good reason not to do it) :) Being able to convert between hparam namespace and yaml files without much workarounds would be very handy

**Comment by Blaizzy** [comment]
Perfect!

Happy it works for you, and the great feedback.

I have passed your feedback to the product team, and they will take it from there.

**Comment by Blaizzy** [comment]
Closing this issue for now.
